libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -715,17 +742,27 @@ void AP_GPS::detect_instance(uint8_t instance)
         _port[instance]->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
         dstate->last_baud_change_ms = now;
 
-        if (_auto_config >= GPS_AUTO_CONFIG_ENABLE_SERIAL_ONLY && new_gps == nullptr) {
+#if AP_GPS_INIT_BLOBS
+        if (_auto_config >= GPS_AUTO_CONFIG_ENABLE_SERIAL_ONLY) {
             send_blob_start(instance);
         }
+#endif
     }
 
-    if (_auto_config >= GPS_AUTO_CONFIG_ENABLE_SERIAL_ONLY && new_gps == nullptr) {
+#if AP_GPS_INIT_BLOBS
```

Comment:
I think this will break hemisphere GPS, which uses send_blob_start(), but AP_GPS_INIT_BLOBS won't be set
I'd prefer to leave the structures in place for the send_blob update, as other drivers may need it in future, and it will become more and more complex to have it in only when needed

Reply:
I've removed the conditional compilation of the blob-sending stuff.